cmd/rook/admission.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/rook/rook/cmd/rook/rook"
+	"github.com/rook/rook/pkg/operator/admission"
+	"github.com/rook/rook/pkg/util/flags"
+	"github.com/spf13/cobra"
+)
+
+var (
+	admissionCmd = &cobra.Command{
+		Use:   "admission-controller",
+		Short: "Starts admission controller",
+	}
+	storageSystem string
+)
+
+func init() {
+	admissionCmd.Flags().StringVar(&storageSystem, "storage", "ceph", "Starts the admission controller for corresponding storage backend. Supported values are: \"ceph\"")
```

Comment:
Instead of the command being `rook admission-controller storage=ceph`, I'm going to suggest we change it to make the admission controller a child of the storage provider so the command would be `rook ceph admission-controller`. The advantages of this approach:
- This clearly implements that one admission controller is implemented per storage provider, if a provider chooses to implement it
- There is a child command for the admission controller, with no arguments necessary
- The call to `admission.StartServer()` on line 47 below would take the callback function such as `cephadmission.ValidateCephResource`, which makes it more strongly typed
- This file would move to `cmd/rook/ceph/admission.go`

Reply:
I have added the command to ceph